### ISSUE TITLE
Handshake Information on Channels

### ISSUE THREAD

**Issue by robmoffat** [finos/FDC3#1131]
## Minor Issue

Here: https://github.com/finos/FDC3/blob/master/docs/agent-bridging/spec.md#step-3-handshake--authentication

There is a field for `channelState`, which tells you what contexts have been broadcast on what channels.

A couple of things:

1.  It doesn't tell you which of these channels are app channels, user channels or private channels - shouldn't that be in the state too?
2.  Surely sending all history is unnecessary, don't you just need the latest version of each context type, in the correct order?

**Comment by kriswest** [comment]
> 1. It doesn't tell you which of these channels are app channels, user channels or private channels - shouldn't that be in the state too?

First off, don't forget the user channels can also be app channels and that private channels are excluded (as per the comment in the docs)! 

It's not ideal but one agent could have different user channels to the other and hence consider that other agent's user channel to be just an app channel. We tried to avoid this by creating a recommended channel set - although I'm not sure if anyone has adopted that yet. We decided to avoid any mapping of channels to alternative Ids for at least the first iteration of bridging to find out if we could avoid it altogether... 

Channels have to have unique ids so that's what we focus on for the state synchronisation and we assume that no one is going to be willing to adopt user channel definitions across the bridge. if they were then we'd need a more detailed data structure for the necessary metadata.

> 2. Surely sending all history is unnecessary, don't you just need the latest version of each context type, in the correct order?

It doesn't send all history, it is literally a case of one context per type - however, the latest context does matter as that's what you'd get if you added an untyped listener to a user channel or called `getCurrentContext` with a null type to an app or private channel. The comment in schema and that dos page could be clearer (feel free to suggest something). I think it is describe more clearly in step 5 as:

> the most recent context object and the most recent context object of each type that it has received

also

> Hence, if we assume that the state of each channel can be represented by an ordered array of context objects (most recent first - noting that only the first position, that of the most recent context broadcast, matters), the Desktop Agent Bridge MUST merge the incoming channelsState with the existingChannelsState as follows

This latter comment could be clearer, how about:

>...represented by an array of the most recent context object of each type, with the most recent context occupying the first position in the array.

If that works it could be copied elsewhere (e.g. step 4, the schema description for the field, etc.)
